src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,46 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder stringBuilder = new StringBuilder("Report for period "
+                + dateFrom + " - " + dateTo);
+        for (String name: names) {
+            int salary = 0;
+            for (String record: data) {
+                if (name.equals(getName(record)) && validateDate(record, dateFrom, dateTo)) {
+                    salary += getDailySalary(record);
+                }
+            }
+            stringBuilder.append(System.lineSeparator()).append(name).append(" - ").append(salary);
+        }
+
+        return stringBuilder.toString();
+    }
+
+    private String getName(String data) {
+        final int namePosition = 1;
+        return data.split(" ")[namePosition];
+    }
+
+    private boolean validateDate(String data, String dateFrom, String dateTo) {
+        final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
```

Comment:
**If you create a [formatter](https://docs.oracle.com/javase/tutorial/datetime/iso/format.html), make it a constant field.**
[Common mistakes](https://mate-academy.github.io/jv-program-common-mistakes/java-core/exceptions/salary-info)